### DISCUSSION TITLE
how prevent delete multiple whitespace on DOMText when exporting excel

### DISCUSSION THREAD

**Discussion by carlosturnerbenites** [PHPOffice/PhpSpreadsheet#2168] [Q&A] [answered]
### What is the expected behavior?
keep multiple whitespace when exporting.
for example with the text "product&nbsp;&nbsp;&nbsp;name" (three spaces between the two words)

### What is the current behavior?
multiple whitespace are not kept
for example "product&nbsp;&nbsp;&nbsp;name" in the exportable is "product name" (without the multiple whitespaces)

### What are the steps to reproduce?
export from html

I was reviewing the documentation and I can't find any configuration to prevent this behavior, is there any configuration to achieve this?

I have also checked the source code of the library a bit and found that the origin of this behavior is:

File: src/PhpSpreadsheet/Reader/Html.php
```
protected function processDomElement(DOMNode $element, Worksheet $sheet, &$row, &$column, &$cellContent) 
{
...
                $domText = preg_replace('/\s+/u', ' ', trim($child->nodeValue));
...
}
```
this line removes multiple whitespace

### Which versions of PhpSpreadsheet and PHP are affected?
Version PHPOffice/PhpSpreadsheet: "1.4.0"
Version PHP: 7.2.34

**Comment by MarkBaker** [comment, marked as answer]
No! There is no configuration to prevent this. This is behaviour that MS Excel itself applies when importing an HTML file
